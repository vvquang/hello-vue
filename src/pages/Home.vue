<template>
  <div class="home-page">
    <h1>Todo List</h1>
    <TodoList />
  </div>
</template>

<script>
const TodoList = () => import('@/components/TodoList')

export default {
  name: 'HomePage',

  components: {
    TodoList
  }
}
</script>
