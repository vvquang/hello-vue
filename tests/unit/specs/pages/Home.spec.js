import { shallowMount } from '@vue/test-utils'
import Home from '@/pages/Home'
import wrapperOps from '../../supports/wrapper'

const wrapper = shallowMount(Home, wrapperOps)

describe('test App.vue', () => {
  it('render app', () => {
    expect(wrapper.exists()).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })
})
