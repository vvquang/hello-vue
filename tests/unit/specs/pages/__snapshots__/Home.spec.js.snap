// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App.vue render app 1`] = `
<div class="home-page">
  <h1>Todo List</h1>
  <div class="todo-list"><input type="text" placeholder="What needs to be done" class="todo-input">
    <ul>
      Todo List goes here:
      <transition-group-stub name="fade" enteractiveclass="animated fadeInUp" leaveactiveclass="animated fadeOutDown">
        <todo-item-stub todo="[object Object]" index="0"></todo-item-stub>
        <todo-item-stub todo="[object Object]" index="1"></todo-item-stub>
      </transition-group-stub>
      <div>
        <div class="extra-container">
          <div><label><input type="checkbox">Check All</label></div>
          <div>1 items left incomplete</div>
        </div>
        <div class="extra-container">
          <div><button class="active">
              All
            </button> <button class="">
              Incomplete
            </button> <button class="">
              Completed
            </button></div>
          <div>
            <transition-stub name="face"><button>
                Clear Completed
              </button></transition-stub>
          </div>
        </div>
      </div>
    </ul>
  </div>
</div>
`;
